pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -202,19 +242,52 @@ func (configgen *ConfigGeneratorImpl) BuildDeltaClusters(proxy *model.Proxy, pus
 		model.XdsLogDetails{AdditionalInfo: fmt.Sprintf("cached:%v/%v", clusterCacheStats.hits, clusterCacheStats.hits+clusterCacheStats.miss)}, true
 }
 
+// getRemovedClusterName finds the name of the removed cluster given a hostname and list of cluster names.
+// subsets is ignored if it is nil
+func getRemovedClusterNames(resNames []string, subsets []string, target string) []string {
+	removed := make([]string, 0)
+	for _, n := range resNames {
```

Comment:
I think this loop is not efficient

Reply:
I mean, watchedResources is not very large and this is O(n)

maybe a `map[host.Name]SubsetValue` where `SubsetValue` is
```go
type SubsetValue struct {
  direction TrafficDirection
  subset string
  host host.Name
  port int
}
```
and we populate/modify whenever watchedresources changes?

Reply:
Yes it is O(n), but multiplex the number of proxies , it is O(n*n)

Reply:
can address optimization in follow up